### PR TITLE
[IMP] product,website_sale(_product_configurator): allow extra-options

### PR DESCRIPTION
**Pull request by vava-odoo**
[A previous commit](https://github.com/odoo/odoo/commit/5ca1da8ab33d4dd6a9ed0b6737e5d19f1a3ac9d2) introduces a new attribute `display_type`, `multi`, which allows users to define extra-options to products. But this option was only available in Sales.

This commit allows eCommerce users to use this new attribute in the online shop.

task-3497046

### CODE REVIEW

File: addons/product/models/product_template.py [hunk 1]
```diff
@@ -949,16 +949,17 @@ def _is_combination_possible_by_config(self, combination, ignore_no_variant=Fals
...
             attribute_lines = attribute_lines._without_no_variant_attributes()
 
-        if len(combination) != len(attribute_lines):
+        if len(combination) != len(attribute_lines) and not has_multi_checkbox:
             # number of attribute values passed is different than the
             # configuration of attributes on the template
             return False
 
-        if attribute_lines != combination.attribute_line_id:
+        if attribute_lines != combination.attribute_line_id and not has_multi_checkbox:
```
Comment (Feyensv) [inline addons/product/models/product_template.py:962]: Should still be a valid check no?

File: addons/product/models/product_template.py [hunk 2]
```diff
@@ -949,16 +949,17 @@ def _is_combination_possible_by_config(self, combination, ignore_no_variant=Fals
         self.ensure_one()
 
         attribute_lines = self.valid_product_template_attribute_line_ids
+        has_multi_checkbox = any(a.display_type == 'multi' for a in attribute_lines.attribute_id)
 
         if ignore_no_variant:
             attribute_lines = attribute_lines._without_no_variant_attributes()
 
-        if len(combination) != len(attribute_lines):
+        if len(combination) != len(attribute_lines) and not has_multi_checkbox:
```
Comment (Feyensv) [inline addons/product/models/product_template.py:957]:
```suggestion
        if len(combination_without_multi) != len(attribute_lines_without_multi):
```
to keep the same level of checks even if there are multi attribute lines ?

File: addons/product/models/product_template.py [hunk 3]
```diff
@@ -1189,6 +1190,16 @@ def _cartesian_product(self, product_template_attribute_values_per_line, parent_
...
             current_ptav_index = value_index_per_line[line_index]
+
+            # For multi-checkbox attribute, the list is empty as we want to start without any selected value
+            if not current_line_values:
+                if line_index == len(product_template_attribute_values_per_line) - 1:
+                    # submit combination if we're on the last line
+                    yield partial_combination
+                else:
+                    line_index += 1
+                    continue
```
Comment (Feyensv) [inline addons/product/models/product_template.py:1201]: I don't see any check here to make sure we're in a multi-checkbox case

Comment (vava-odoo) [reply addons/product/models/product_template.py:1201]: `current_line_values` is a list of available ptavs, but is empty (only) for 'multi' (see https://github.com/odoo/odoo/pull/135174/files#diff-921ec733c3821301f62cd0d96e45990e7636f5db56a10ae76aed009e380bf149R1284)

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by Feyensv** [comment]
@robodoo r+

**Pull request closed without merging**